sdk/src/trace/batch_span_processor.cc

```diff
@@ -175,12 +175,14 @@ void BatchSpanProcessor::DrainQueue()
 
 void BatchSpanProcessor::Shutdown(std::chrono::microseconds timeout) noexcept
 {
-  is_shutdown_ = true;
+  is_shutdown_.store(true);
```

Comment:
@ThomsonTan - I think we may want to add a safety check here, such as:

```cpp
if (!is_shutdown.exchange(true))
{
  cv_.notify_one();
  worker_thread_.join();
  if (exporter_ != nullptr)
  {
    exporter_->Shutdown();
  }
}
```

Otherwise we may run into C++ exception at `join()` with `system_error` in case if it's been already joined. Do we need to make it more forgiving?